lib/stdlib/src/dets.erl

```diff
@@ -1237,15 +1238,27 @@ treq(Tab, R) ->
 
 req(Proc, R) ->
     Ref = erlang:monitor(process, Proc),
-    Proc ! ?DETS_CALL(self(), R),
+    Proc ! ?DETS_CALL({self(), Ref}, R),
     receive 
 	{'DOWN', Ref, process, Proc, _Info} ->
             badarg;
-	{Proc, Reply} ->
+	{Proc, Ref, Reply} ->
 	    erlang:demonitor(Ref, [flush]),
 	    Reply
     end.
 
+%% Inlined.
+pidof({Pid, _Ref}) ->
+    Pid;
+pidof(Pid) ->
+    Pid.
+
+%% Inlined.
+resp({Pid, Ref} = _From, Message) ->
+    Pid ! {self(), Ref, Message}, ok;
+resp(Pid, Message) ->
+    Pid ! {self(), Message}, ok.
+
```

Comment:
Do you really need to keep support for replying to old Ref-less calls?

Reply:
You are right, we do not need to keep it anymore. Fixed.

Reply:
`dets` server sends some messages to itself without using references (`auto_save` | `may_grow` | `{delayed_write, _}`) but in that case it does not send nor expect any response.